### PR TITLE
AEA-570 fix and align tests setup

### LINKED ISSUE #1164: Move all github workflow commands to make file to call it localy
**Is your feature request related to a problem? Please describe.**
`make test` does not correspond completely to Github workflow ci checks.
to check locally I have to copy-paste some steps from github workflow scenario.

**Describe the solution you'd like**
It would be nice to have the same scenario to make tests locally as it is done using Github workflow.
Every step of github workflow can be implemented in make file, and git workflow will call it like `make ci_py3.8`, I can do the same locally and get near the same results.

**Describe alternatives you've considered**
probably https://github.com/nektos/act will be the best solution but requires bulky image
and does not support targets except ubuntu

**Additional context**
It will help test it locally in the same way as CI does.

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR aligns our test setup.

## Fixes

#1164

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -31,10 +31,8 @@ jobs:
            tar xvfz go-ipfs.tar.gz
            sudo mv go-ipfs/ipfs /usr/local/bin/ipfs
            ipfs init
-       - name: Security Check - Main
-         run: tox -e bandit-main
-       - name: Security Check - Tests
-         run: tox -e bandit-tests
+       - name: Security Check
+         run: tox -e bandit
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:35]: The previous split was now redundant

File: Makefile [hunk 2]
```diff
@@ -45,14 +45,28 @@ lint:
 
 .PHONY: security
 security:
-	bandit -s B101 -r aea packages scripts
-	bandit -s B101 -r tests
-	bandit -s B101 -r benchmark
-	safety check
+	bandit -s B101 -r aea benchmark examples packages scripts tests
+	safety check -i 37524 -i 38038 -i 37776 -i 38039
```
Comment (DavidMinarsch) [inline Makefile:49]: these ignore issues related with tensorflow

File: Makefile [hunk 3]
```diff
@@ -45,14 +45,28 @@ lint:
...
 static:
-	mypy aea benchmark packages tests scripts
+	mypy aea benchmark examples packages scripts tests
+
+.PHONY: misc_checks
+misc_checks:
+	# pip install '.[all]'
+	# python scripts/freeze_dependencies.py -o requirements.txt
+	# liccheck -s strategy.ini -r requirements.txt -l PARANOID
+	# rm -fr requirements.txt
```
Comment (DavidMinarsch) [inline Makefile:60]: @marcofavorito I cannot get this to work locally..it claims due to editable ...

Comment (marcofavorito) [reply Makefile:60]: `liccheck` works by checking the packages installed in the current environment.

You have to create a temporary virtual environment and install the `aea` package, like it's done in `tox -e liccheck`.

Comment (solarw) [reply Makefile:60]: @DavidMinarsch 
do we really need to check tests with mypy?

Comment (DavidMinarsch) [reply Makefile:60]: Yes 100% mypy everywhere.

**Pull request merged**